File: HRM_systemAnguler/src/app/services/country.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Country } from '../models/country';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
@Injectable({
  providedIn: 'root',
})
export class CountryService {
  private apiUrl = `${environment.baseUrl}/api/country`;
  constructor(private http: HttpClient) {}
  saveCountry(country: Country): Observable<Country> {
    return this.http.post<Country>(`${this.apiUrl}/save`, country);
  }
  getAllCountries(): Observable<Country[]> {
    return this.http.get<Country[]>(`${this.apiUrl}/all`);
  }
  getCountryById(id: number): Observable<Country> {
    return this.http.get<Country>(`${this.apiUrl}/${id}`);
  }
  getCountryByName(name: string): Observable<Country> {
    return this.http.get<Country>(`${this.apiUrl}/name/${name}`);
  }
  updateCountry(id: number, country: Country): Observable<Country> {
    return this.http.put<Country>(`${this.apiUrl}/update/${id}`, country);
  }
  deleteCountry(id: number): Observable<string> {
    return this.http.delete(`${this.apiUrl}/delete/${id}`, {
      responseType: 'text',
    });
  }
}

File: HRM_system/src/main/java/com/emranhss/HRM_system/location/country/CountryService.java
package com.emranhss.HRM_system.location.country;

import org.springframework.stereotype.Service;

import java.util.List;
@Service
public interface CountryService {
    
    CountryResponseDto saveCountry(CountryRequestDto dto);

    
    CountryResponseDto getCountryById(Long id);

    
    List<CountryResponseDto> getAllCountries();

    
    CountryResponseDto updateCountry(Long id,
                                     CountryRequestDto dto);

    
    void deleteCountry(Long id);

    
    CountryResponseDto getCountryByName(String countryName);

}
